pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -235,6 +235,19 @@ func ValidateFlowSchemaResourcePolicyRule(rule *flowcontrol.ResourcePolicyRule,
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resources"), rule.Resources, "if '*' is present, must not specify other resources"))
 	}
 
+	if len(rule.Namespaces) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("namespaces"), "resource rules must supply at least one namespace"))
+	} else {
+		for idx, tgtNS := range rule.Namespaces {
+			if tgtNS == flowcontrol.NamespaceEvery || tgtNS == flowcontrol.NamespaceNot {
+				continue
+			}
+			for _, msg := range apimachineryvalidation.ValidateNamespaceName(tgtNS, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("namespaces").Index(idx), tgtNS, msg))
+			}
+		}
```

Comment:
We should require that if "*" is present, one mustn't also reference specific namespaces, although it's OK to have "Cluster Scope" too.